conans/client/tools/settings.py

```diff
@@ -0,0 +1,63 @@
+from conans.errors import ConanInvalidConfiguration
+from conans.client.build.cppstd_flags import cppstd_flag, cppstd_from_settings
+from conans.client.tools.oss import OSInfo
+
+
+def check_min_cppstd(conanfile, cppstd, gnu_extensions=False):
+    """ Check if current cppstd fits the minimal version required.
+
+        In case the current cppstd doesn't fit the minimal version required
+        by cppstd, a ConanInvalidConfiguration exception will be raised.
+
+    :param conanfile: ConanFile instance with cppstd to be compared
+    :param cppstd: Minimal cppstd version required
+    :param gnu_extensions: GNU extension is required (e.g gnu17). This option ONLY works on Linux.
+    """
+    assert (cppstd is not None), "Cannot check invalid cppstd version"
```

Comment:
Do not use asserts for "production" code, use regular Exceptions. Proof that it is "production" code is that you are testing this behavior in the tests.

Reply:
Done, changed to ConanException